src/lib.rs

```diff
@@ -0,0 +1,414 @@
+use cratesio_dbdump_csvtab::rusqlite::{self, Connection, Error};
+use std::num::ParseIntError;
+use tracing::instrument;
+
+pub trait CrateFetcher {
+    fn get_crate(&self, crate_name: &str) -> Result<Option<Crate>, rusqlite::Error>;
+    fn get_keywords(&self, crate_id: &str) -> Result<Vec<String>, rusqlite::Error>;
+}
```

Comment:
Since this is a trait in a library, maybe we should return our own errors instead of rusqlite ones 🤔

Reply:
Notably the other lib already uses thiserror for it.

Reply:
This is temp atm. plus I don't know what the best way is to do so either. Feel free to give it a go.

Reply:
Alright I'll experiment with it later, prob gonna merge now and look into publishing a 0.1 version if you are okay with that.

Reply:
Cool cool 👍, I think #85f97af is gonna be my last commit for today 😅